src/Features/Core/Portable/Workspace/ProjectCacheService.cs

```diff
@@ -30,18 +32,20 @@ internal partial class ProjectCacheService : IProjectCacheHostService
         private readonly Workspace _workspace;
         private readonly Dictionary<ProjectId, Cache> _activeCaches = new Dictionary<ProjectId, Cache>();
 
+        private readonly CancellationTokenSource _shutdownNotificationSource = new();
```

Comment:
why a CTS of its own, instead of using ITHreadingContext.DisposalToken?

Reply:
➡️ This is covered by the discussion in https://github.com/dotnet/roslyn/pull/47902#discussion_r492389637